### PR TITLE
Support HTTP protocol

### PR DESCRIPTION
**Pull request by stefanseifert**
Surprisingly the current implementation only supports HTTPs protocol, no HTTP protocol (HTTP client throws an error if HTTP protocl is used). For local development it's useful to also support HTTP.

This PR adds support for HTTP protocol including test coverage (with some test refactoring to re-use code from ConcurrencyTest).

### CODE REVIEW

**Comment by cjelger** [comment]
Thx @stefanseifert but the AEM security team disallowed us a while ago to support HTTP so this is an intentional limitation.

**Comment by stefanseifert** [comment]
thanks for the quick response - does it really make sense to hardcode such a limitation in the code?
it unnecessary complicates local development.
if this cannot be changed it at least should be documented (and commented in the code) - in the current state it really looked like an oversight, therefore i've provided the PR.

**Comment by cjelger** [comment]
I'll check again with the security team, and yes, it does complicate local development. If we are not allowed to support HTTP, I'll make sure we add a note in the documentation and code.

**Comment by stefanseifert** [comment]
maybe an alternative would be to add an OSgi configuration option "allow HTTP" same as "allow self-signed certificates" which produces a warning when it it switched on.

**Comment by acollign** [comment]
I understand the need to ease local development. I recommend implementing an option `insecure`* option which allows HTTP connections. Please ensure:
- HTTP support is off by default
- risk is clearly documented
- Enabling HTTP support logs WARN

* think of curl --insecure

**Comment by stefanseifert** [comment]
i've updated the PR accordingly, including test coverage for HTTP allowed/disallowed

File: src/test/java/com/adobe/cq/commerce/graphql/client/impl/ConcurrencyTest.java [hunk 1]
```diff
@@ -122,23 +92,19 @@ public void testConcurrencyWithErrors() throws Exception {
         // If the error messages are not consumed by the client, this test will time out
         // because all threads from the connection pool will stay stuck
 
-        HttpResponse error = new HttpResponse()
-            .withStatusCode(INTERNAL_SERVER_ERROR_500.code())
+        HttpResponse error = new HttpResponse().withStatusCode(INTERNAL_SERVER_ERROR_500.code())
```
Comment (cjelger) [inline src/test/java/com/adobe/cq/commerce/graphql/client/impl/ConcurrencyTest.java:95]: I'm being picky, but you should avoid reformatting code you don't change, it unnecessarily "pollute" the PR with unrelated changes.

**Comment by mhaack** [comment]
Verified this today using AEM Cloud SDK and Venia connected to Magento via `http`
